packages/data-stores/src/products-list/resolvers.ts

```diff
@@ -0,0 +1,24 @@
+import { dispatch } from '@wordpress/data';
+import { wpcomRequest } from '../wpcom-request-controls';
+import { STORE_KEY } from './constants';
+import { ProductsList, ProductsListFailure } from './types';
+
+/**
+ * Requests the list of all products from the WPCOM API.
+ *
+ * @param   {string} type The type of products to request (e.g., "jetpack");
+ */
+export function* getProductsList( type: string | null ) {
+	yield dispatch( STORE_KEY ).requestProductsList();
```

Comment:
The store does the dispatching automatically:

```suggestion
	yield requestProductsList();
```

Reply:
I've been mostly using the `site` data-store as a guide for new functionality and it's still using `dispatch( STORE_KEY )`.

It's good to know that can be simplified.